apps/gdalalg_raster_blend.cpp

```diff
@@ -675,70 +729,251 @@ CPLErr HSVMergeBand::IRasterIO(GDALRWFlag eRWFlag, int nXOff, int nYOff,
+                    ? m_oBlendDataset.m_abyBuffer.data() +
+                          nPixelCount * (nColorCount + 1)
+                    : nullptr;
+            const GByte *pabyOverlayB =
+                (nOverlayCount >= 3 && nColorCount < 3 && nBand <= 3)
+                    ? m_oBlendDataset.m_abyBuffer.data() +
+                          nPixelCount * (nColorCount + 2)
+                    : nullptr;
+
+            size_t nSrcIdx = 0;
+            for (int j = 0; j < nBufYSize; ++j)
+            {
+                auto nDstOffset = j * nLineSpace;
+                for (int i = 0; i < nBufXSize;
+                     ++i, ++nSrcIdx, nDstOffset += nPixelSpace)
+                {
+                    const int nOverlayA =
+                        pabyOverlayA ? ((pabyOverlayA[nSrcIdx] *
+                                             m_oBlendDataset.m_opacity255Scale +
+                                         255) /
+                                        256)
+                                     : m_oBlendDataset.m_opacity255Scale;
+
+                    const int nSrc = paby ? paby[nSrcIdx] : 255;
+
+                    const int nOverlay =
+                        (pabyOverlayR && pabyOverlayG && pabyOverlayB)
+                            ? RGBToGrayScale(pabyOverlayR[nSrcIdx],
+                                             pabyOverlayG[nSrcIdx],
+                                             pabyOverlayB[nSrcIdx])
+                        : pabyOverlay ? pabyOverlay[nSrcIdx]
+                                      : 255;
+
+                    pabyDst[nDstOffset] = static_cast<GByte>(
+                        std::min((nOverlay * m_oBlendDataset.m_opacity255Scale +
+                                  nSrc * (255 - nOverlayA) + 255) /
+                                     256,
+                                 255));
+                }
             }
-            else
+        }
+        else if (nOverlayCount == 1 && m_oBlendDataset.m_opacity255Scale == 255)
+        {
+            const GByte *pabyR = m_oBlendDataset.m_abyBuffer.data();
+            const GByte *pabyG =
+                m_oBlendDataset.m_abyBuffer.data() + nPixelCount;
+            const GByte *pabyB =
+                m_oBlendDataset.m_abyBuffer.data() + nPixelCount * 2;
+            CPLAssert(m_oBlendDataset.m_operator == HSV_VALUE);
+            size_t nSrcIdx = 0;
+            const GByte *pabyValue =
+                m_oBlendDataset.m_abyBuffer.data() + nPixelCount * nColorCount;
+            for (int j = 0; j < nBufYSize; ++j)
             {
+                auto nDstOffset = j * nLineSpace;
+                if (nPixelSpace == 1 && nLineSpace >= nPixelSpace * nBufXSize)
+                {
+                    patch_value_line(
+                        nBufXSize, pabyR + nSrcIdx, pabyG + nSrcIdx,
+                        pabyB + nSrcIdx, pabyValue + nSrcIdx,
+                        nBand == 1 ? pabyDst + nDstOffset : nullptr,
```

Comment:
I'm not sure this is correct: is it intended that the three output pointers (R, G and B) have the same value (it they are not null)? Shouldn't they be separated by offset?